jack.py

```diff
@@ -187,6 +187,35 @@
 void jack_set_info_function(void (*func)(const char*));
 void jack_free(void* ptr);
 
+/* ringbuffer.h */
+typedef struct {
+    char *buf;
+    size_t len;
+} jack_ringbuffer_data_t;
+typedef struct {
+    char *buf;
+    volatile size_t write_ptr;
+    volatile size_t read_ptr;
+    size_t size;
+    size_t size_mask;
+    int mlocked;
+} jack_ringbuffer_t ;
+
+jack_ringbuffer_t *jack_ringbuffer_create(size_t sz);
+void jack_ringbuffer_free(jack_ringbuffer_t *rb);
+/* not implemented: jack_ringbuffer_get_read_vector, really fast uses only */
+/* not implemented: jack_ringbuffer_get_write_vector, really fast uses only */
+size_t jack_ringbuffer_read(jack_ringbuffer_t *rb, char *dest, size_t cnt);
+size_t jack_ringbuffer_peek(jack_ringbuffer_t *rb, char *dest, size_t cnt);
+void jack_ringbuffer_read_advance(jack_ringbuffer_t *rb, size_t cnt);
+size_t jack_ringbuffer_read_space(const jack_ringbuffer_t *rb);
+int jack_ringbuffer_mlock(jack_ringbuffer_t *rb);
+void jack_ringbuffer_reset(jack_ringbuffer_t *rb);
+/* not implemented: jack_ringbuffer_reset_size */
```

Comment:
Why not implement this?

This could be included in `reset()` by an optional size parameter: `def reset(self, size=None)`.